pypose/function/geometry.py

```diff
@@ -159,3 +161,122 @@ def reprojerr(points, pixels, intrinsics, extrinsics=None):
            intrinsics.size(-1) == intrinsics.size(-2) == 3, "Shape not compatible."
     img_repj = point2pixel(points, intrinsics, extrinsics)
     return (img_repj - pixels).norm(dim=-1)
+
+
+def knn(pc1, pc2, k = 1, norm = 2):
```

Comment:
Don’t use spaces around the = sign when used to indicate a keyword argument
See https://peps.python.org/pep-0008/#other-recommendations